diff --git a/src/nhflow_kepsilon_func.cpp b/src/nhflow_kepsilon_func.cpp
index 1e6797eaa..7e500e4eb 100644
--- a/src/nhflow_kepsilon_func.cpp
+++ b/src/nhflow_kepsilon_func.cpp
@@ -84,7 +84,9 @@ void nhflow_kepsilon_func::eddyvisc(lexer* p, fdm_nhf *d, ghostcell* pgc, vrans*
 	double H;
 	int n;
         
-        
+    // RANS
+    if(p->A560==1)
+    {
         if(p->A564==0)
         LOOP
 		d->EV0[IJK] = p->cmu*MAX(MAX(KIN[IJK]*KIN[IJK]
@@ -109,9 +111,10 @@ void nhflow_kepsilon_func::eddyvisc(lexer* p, fdm_nhf *d, ghostcell* pgc, vrans*
 						  /((EPS[IJK])>(1.0e-20)?(EPS[IJK]):(1.0e20)),0.0),fabs(p->T35*KIN[IJK])/strainterm(p,d)),
 						  0.00001*d->VISC[IJK]);
 		}
+    }
 	
     // URANS
-	if(p->A560==22)
+	if(p->A560==21)
 	LOOP
     {
     if(p->j_dir==0)
@@ -120,17 +123,17 @@ void nhflow_kepsilon_func::eddyvisc(lexer* p, fdm_nhf *d, ghostcell* pgc, vrans*
     if(p->j_dir==1)
     dxm = pow(p->DXN[IP]*p->DYN[JP]*p->DZN[KP]*d->WL(i,j), (1.0/3.0));
     
-	f = MIN(1.0, dxm*p->A568*p->cmu*EPS[IJK]/   pow((KIN[IJK]>(1.0e-20)?(KIN[IJK]):(1.0e20)),0.5));
+	f = MIN(1.0, dxm*p->A568*EPS[IJK]/pow((KIN[IJK]>(1.0e-20)?(KIN[IJK]):(1.0e20)),1.5));
     
     
     
         if(p->A564==0)
-		d->EV0[IJK] = f*MAX(MAX(KIN[IJK]*KIN[IJK]
+		d->EV0[IJK] = p->cmu*f*MAX(MAX(KIN[IJK]*KIN[IJK]
 						  /((EPS[IJK])>(1.0e-20)?(EPS[IJK]):(1.0e20)),0.0),
 						  0.00001*d->VISC[IJK]);
                           
         if(p->A564==1)
-		d->EV0[IJK] = f*MAX(MIN(MAX(KIN[IJK]*KIN[IJK]
+		d->EV0[IJK] = p->cmu*f*MAX(MIN(MAX(KIN[IJK]*KIN[IJK]
 						  /((EPS[IJK])>(1.0e-20)?(EPS[IJK]):(1.0e20)),0.0),fabs(p->T31*KIN[IJK])/strainterm(p,d)),
 						  0.00001*d->VISC[IJK]);
                           
diff --git a/src/nhflow_komega_func.cpp b/src/nhflow_komega_func.cpp
index c67b0bea7..bbfef6c0d 100644
--- a/src/nhflow_komega_func.cpp
+++ b/src/nhflow_komega_func.cpp
@@ -83,8 +83,10 @@ void nhflow_komega_func::eddyvisc(lexer* p, fdm_nhf *d, ghostcell* pgc, vrans* p
 	double factor;
 	double H;
 	int n;
-        
-        
+    
+    // RANS
+    if(p->A560==2)
+    {
         if(p->A564==0)
         LOOP
 		d->EV0[IJK] = MAX(KIN[IJK]
@@ -110,7 +112,8 @@ void nhflow_komega_func::eddyvisc(lexer* p, fdm_nhf *d, ghostcell* pgc, vrans* p
 						  /((EPS[IJK])>(1.0e-20)?(EPS[IJK]):(1.0e20)),0.0),fabs(p->T35*KIN[IJK])/strainterm(p,d)),
 						  0.00001*d->VISC[IJK]);
 		}
-	
+	}
+    
     // URANS
 	if(p->A560==22)
 	LOOP
diff --git a/src/nhflow_timeavg_vel_profile.cpp b/src/nhflow_timeavg_vel_profile.cpp
index b97ed2e1a..9b7795a4f 100644
--- a/src/nhflow_timeavg_vel_profile.cpp
+++ b/src/nhflow_timeavg_vel_profile.cpp
@@ -177,7 +177,7 @@ void nhflow_timeavg_vel_profile::ini_location(lexer *p, fdm_nhf *d)
     {
         iloc[n]=p->posc_i(p->P148_x[n]);
 
-        if(p->j_dir==0)
+        if(p->j_dir==0 || p->knoy==1)
         {
             jloc[n]=0;
             p->P148_y[n] = 0.5*p->YP[JP];
