diff --git a/src/wave_lib_spectrum_TMA.cpp b/src/wave_lib_spectrum_TMA.cpp
index 04ca9e633..e8382d4e1 100644
--- a/src/wave_lib_spectrum_TMA.cpp
+++ b/src/wave_lib_spectrum_TMA.cpp
@@ -39,15 +39,19 @@ double wave_lib_spectrum::TMA(lexer *p, double w)
     Sval *= (1.0-0.287*log(p->B88))*pow(p->B88,exp(-0.5*pow(((w-p->wwp)/(sigma*p->wwp)),2.0)));
 
     // TMA
+    double wdt_tma;
+    wdt_tma=p->F60;
+    if(p->B94==1)
+    wdt_tma=p->B94_wdt;
 
     double phi_tma;
-    if(w*sqrt(p->F60/9.81)<1)
-    phi_tma=0.5*pow(w*sqrt(p->F60/9.81),2);
+    if(w*sqrt(wdt_tma/9.81)<1)
+    phi_tma=0.5*pow(w*sqrt(wdt_tma/9.81),2);
 
-    if(w*sqrt(p->F60/9.81)>=1 && w*sqrt(p->F60/9.81)<2)
-    phi_tma=(1-0.5*pow(2-w*sqrt(p->F60/9.81),2));
+    if(w*sqrt(wdt_tma/9.81)>=1 && w*sqrt(wdt_tma/9.81)<2)
+    phi_tma=(1-0.5*pow(2-w*sqrt(wdt_tma/9.81),2));
 
-    if(w*sqrt(p->F60/9.81)>=2)
+    if(w*sqrt(wdt_tma/9.81)>=2)
     phi_tma=1.0;
 
     Sval *= phi_tma;
